server/src/test/java/org/candlepin/model/PoolCuratorTest.java

```diff
@@ -1464,6 +1462,78 @@ public void testGetPoolsByFilter() {
         assertThat(results, Matchers.hasItems(p1Attributes, p2Attributes));
     }
 
+
+    @Test
+    public void testGetPoolsOrderedByProductNameAscending() {
+
+        Owner owner1 = this.createOwner();
+        this.ownerCurator.create(owner1);
+
+        Pool p1 = TestUtil.createPool(owner1, this.generateProduct(owner1, "p1", "xyz"));
+        p1.setSourceSubscription(new SourceSubscription("subscriptionId-phil", "master"));
+        Pool p2 = TestUtil.createPool(owner1, this.generateProduct(owner1, "p2", "abc"));
+        p2.setSourceSubscription(new SourceSubscription("subscriptionId-ned", "master1"));
+        Pool p3 = TestUtil.createPool(owner1, this.generateProduct(owner1, "p3", "lmn"));
+        p3.setSourceSubscription(new SourceSubscription("subscriptionId-ned1", "master11"));
+        this.poolCurator.create(p3);
+        this.poolCurator.create(p1);
+        this.poolCurator.create(p2);
+        PageRequest req = new PageRequest();
+
+        req.setOrder(PageRequest.Order.ASCENDING);
+        req.setSortBy("Product.name");
+        Date activeOn = TestUtil.createDate(2019, 2, 2);
+
+
+        Page<List<Pool>> page = poolManager.listAvailableEntitlementPools(null, null, owner1.getId(), null, null, activeOn, false, null,
+                req, false, false, null);
+
+        List<Pool> pools = page.getPageData();
+        List<Pool> results = new ArrayList<>();
+        results.add(p2);
+        results.add(p3);
+        results.add(p1);
+        assertEquals(results, pools);
+
+    }
+
+    @Test
+    public void testGetPoolsOrderedByProductNameDescending() {
+
+        //Checking for Descending
+        Owner owner1 = this.createOwner();
+        this.ownerCurator.create(owner1);
+
+        Pool p1 = TestUtil.createPool(owner1, this.generateProduct(owner1, "p1", "xyz"));
+        p1.setSourceSubscription(new SourceSubscription("subscriptionId-phil", "master"));
+        Pool p2 = TestUtil.createPool(owner1, this.generateProduct(owner1, "p2", "abc"));
+        p2.setSourceSubscription(new SourceSubscription("subscriptionId-ned", "master1"));
+        Pool p3 = TestUtil.createPool(owner1, this.generateProduct(owner1, "p3", "lmn"));
+        p3.setSourceSubscription(new SourceSubscription("subscriptionId-ned1", "master11"));
+        this.poolCurator.create(p3);
+        this.poolCurator.create(p1);
+        this.poolCurator.create(p2);
+        PageRequest req1 = new PageRequest();
+        req1.setSortBy("Product.name");
+        req1.setOrder(PageRequest.Order.DESCENDING);
+
+        Date activeOn1 = TestUtil.createDate(2019, 2, 2);
+
+        Page<List<Pool>> page1 = poolManager.listAvailableEntitlementPools(null, null,  owner1.getId(), null, null, activeOn1, false, null,
```

Comment:
This line (and the other matching one in the new test) is too long for the line limits.

The gradle linting tests appear to be broken, so for the interim, make sure you run the following command to check code style:
```buildr checkstyle```

Reply:
./gradlew check is showing the error regarding line length

Reply:
I was wrong, the cleanest way to ensure you see the issues is "./gradlew checkstyleTest checkstyleMain", I submitted a separate PR to update the the docker/jenkins script to run both targets.